src/mscorlib/src/System/Runtime/Intrinsics/X86/Sse2.cs

```diff
@@ -1023,47 +1024,90 @@ public static class Sse2
+        }
         /// <summary>
         /// __m128i _mm_set1_epi16 (short a)
         ///   HELPER
         /// </summary>
-        public static Vector128<ushort> SetAllVector128(ushort value) => SetAllVector128(value);
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector128<ushort> SetAllVector128(ushort value)
+        {
+            Vector128<ushort> vector = Sse.StaticCast<uint, ushort>(ConvertScalarToVector128UInt32(value));
+            return Sse.StaticCast<uint, ushort>(Shuffle(Sse.StaticCast<ushort,uint>(UnpackLow(vector, vector)), (byte)0));
+        }
         /// <summary>
         /// __m128i _mm_set1_epi32 (int a)
         ///   HELPER
         /// </summary>
-        public static Vector128<int> SetAllVector128(int value) => SetAllVector128(value);
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector128<int> SetAllVector128(int value)
+        {
+            return Shuffle(ConvertScalarToVector128Int32(value), 0);
+        }
         /// <summary>
         /// __m128i _mm_set1_epi32 (int a)
         ///   HELPER
         /// </summary>
-        public static Vector128<uint> SetAllVector128(uint value) => SetAllVector128(value);
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector128<uint> SetAllVector128(uint value)
+        {
+            return Shuffle(ConvertScalarToVector128UInt32(value), 0);
+        }
         /// <summary>
         /// __m128i _mm_set1_epi64x (long long a)
         ///   HELPER
         /// </summary>
-        public static Vector128<long> SetAllVector128(long value) => SetAllVector128(value);
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector128<long> SetAllVector128(long value)
+        {
+            return Sse.StaticCast<int, long>(Shuffle(Sse.StaticCast<long, int>(ConvertScalarToVector128Int64(value)), (byte)0b01000100));
+        }
         /// <summary>
         /// __m128i _mm_set1_epi64x (long long a)
         ///   HELPER
         /// </summary>
-        public static Vector128<ulong> SetAllVector128(ulong value) => SetAllVector128(value);
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector128<ulong> SetAllVector128(ulong value)
+        {
+            return Sse.StaticCast<int, ulong>(Shuffle(Sse.StaticCast<ulong, int>(ConvertScalarToVector128UInt64(value)), (byte)0b01000100));
+        }
         /// <summary>
         /// __m128d _mm_set1_pd (double a)
         ///   HELPER
         /// </summary>
-        public static Vector128<double> SetAllVector128(double value) => SetAllVector128(value);
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector128<double> SetAllVector128(double value)
+        {
+            Vector128<double> vector = SetScalarVector128(value);
+            return Shuffle(vector, vector, 0);
```

Comment:
`UnpackLow` will result in slightly smaller codegen and will do the same thing (instructions are listed at same latency/throughput).

Reply:
Will update